### PR TITLE
Update code to return the correct error message in cases of uninitialised spreadsheets

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

Fixes:[4505](https://github.com/ballerina-platform/ballerina-standard-library/issues/4505)

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [x] Updated the changelog
- [x] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 ### Changed
+- [Update code to return the correct error message in cases of uninitialised spreadsheets](https://github.com/ballerina-platform/ballerina-standard-library/issues/4505)
```
Comment (kaneeldias) [inline changelog.md]: I think we don't need the "Update code to return" phrase in the changelogs. Just  "Updated error message in cases of uninitialized spreadsheets" is better.

@daneshk wdyt?

**Review by kaneeldias — DISMISSED** [review]

File: ballerina/googlesheets_client.bal [hunk 2]
```diff
@@ -185,9 +185,15 @@ public isolated client class GoogleSheetsClient {
                 int i = 0;
                 record {} rowArray = {};
                 string[] rowValues = re `,`.split(rowString);
+                if columnNames.length() != self.dataTypes.length() {
+                    return <persist:Error>error("Error: the spreadsheet is not initialised correctly.");
```
Comment (daneshk) [inline ballerina/googlesheets_client.bal]: Shall we say why it is not initialized? `sheets columns do not match with the field types`

File: ballerina/googlesheets_client.bal [hunk 3]
```diff
@@ -185,9 +185,15 @@ public isolated client class GoogleSheetsClient {
...
+                }
                 foreach string rowValue in rowValues {
                     string columnName = re ` `.replaceAll(re `"`.replaceAll(columnNames[i], ""), "");
                     string value = re `"`.replaceAll(rowValue, "");
+                    if !self.dataTypes.hasKey(columnName) {
+                        return <persist:Error>error("Error: the spreadsheet is not initialised correctly.");
```
Comment (daneshk) [inline ballerina/googlesheets_client.bal]: same as above

File: changelog.md (hunk 1 above)
Comment (sahanHe) [reply changelog.md]: updated in [7cc2d06](https://github.com/ballerina-platform/module-ballerinax-persist.googlesheets/pull/38/commits/7cc2d06b018923a1e05a4f3cd1f8961834b48bcd)

File: ballerina/googlesheets_client.bal (hunk 2 above)
Comment (sahanHe) [reply ballerina/googlesheets_client.bal]: updated in [7cc2d06](https://github.com/ballerina-platform/module-ballerinax-persist.googlesheets/pull/38/commits/7cc2d06b018923a1e05a4f3cd1f8961834b48bcd)

File: ballerina/googlesheets_client.bal (hunk 3 above)
Comment (sahanHe) [reply ballerina/googlesheets_client.bal]: updated in [7cc2d06](https://github.com/ballerina-platform/module-ballerinax-persist.googlesheets/pull/38/commits/7cc2d06b018923a1e05a4f3cd1f8961834b48bcd)

**Review by daneshk — APPROVED** [review]

**Pull request merged**